src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -473,6 +539,55 @@ class SwapIFrameApi extends BitcoinRequestParserMixin(RequestParser) { // eslint
             }
         }
 
+        if (parsedRequest.fund.type === 'USDC' && storedRequest.fund.type === 'USDC') {
+            const usdcHtlcContract = new ethers.Contract(
+                CONFIG.USDC_HTLC_CONTRACT_ADDRESS,
+                PolygonContractABIs.USDC_HTLC_CONTRACT_ABI,
+            );
+
+            // Place contract details into existing function call data
+            storedRequest.fund.request.data = usdcHtlcContract.interface.encodeFunctionData(
+                storedRequest.fund.description.name,
+                [
+                    /* bytes32 id */ parsedRequest.fund.description.args.id,
+                    /* address token */ parsedRequest.fund.description.args.token,
+                    /* uint256 amount */ parsedRequest.fund.description.args.amount,
+                    /* address refundAddress */ parsedRequest.fund.description.args.refundAddress,
+                    /* address recipientAddress */ parsedRequest.fund.description.args.recipientAddress,
+                    /* bytes32 hash */ parsedRequest.fund.description.args.hash,
+                    /* uint256 timeout */ parsedRequest.fund.description.args.timeout,
+
+                    /* uint256 fee */ storedRequest.fund.description.args.fee,
+                    ...(storedRequest.fund.description.name === 'openWithApproval' ? [
+                        /* uint256 approval */ storedRequest.fund.description.args.approval,
+                        /* bytes32 sigR */ storedRequest.fund.description.args.sigR,
+                        /* bytes32 sigS */ storedRequest.fund.description.args.sigS,
+                        /* uint8 sigV */ storedRequest.fund.description.args.sigV,
+                    ] : []),
+                ],
+            );
+
+            const typedData = new OpenGSN.TypedRequestData(CONFIG.POLYGON_CHAIN_ID, CONFIG.USDC_HTLC_CONTRACT_ADDRESS, {
+                request: storedRequest.fund.request,
+                relayData: storedRequest.fund.relayData,
+            });
+
+            const { EIP712Domain, ...cleanedTypes } = typedData.types;
+
+            const wallet = new ethers.Wallet(privateKeys.usdc);
+
+            const signature = await wallet._signTypedData(
+                typedData.domain,
+                /** @type {Record<string, ethers.ethers.TypedDataField[]>} */ (/** @type {unknown} */ (cleanedTypes)),
+                typedData.message,
+            );
+
+            result.usdc = {
+                message: typedData.message,
+                signature,
+            };
```

Comment:
Because it has a high likelihood that it will be invalid (i.e. the relay doesn't work anymore or doesn't have enough balance), so I did not implement it (yet). 

We found that in mainnet the relays are more reliable, so makes sense to add it now.